app/routes/signup.js

```diff
@@ -2,27 +2,40 @@ import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
 import ENV from 'website-my/config/environment';
 import { SIGNUP } from '../constants/analytics';
+import {
+  AUTH_URL,
+  ERROR_MESSAGES,
+  GOTO_URL,
+  REDIRECT_TEXT,
+} from '../constants/signup';
 
 export default class SignupRoute extends Route {
   @service analytics;
-
-  async model() {
-    this.analytics.trackEvent(SIGNUP.PAGE_LOADED);
-    const response = await fetch(`${ENV.BASE_API_URL}/users/self`, {
-      credentials: 'include',
-    });
-    const userData = await response.json();
-    if (response.status === 401) {
-      alert('You have not logged in. Please login first to fill this form.');
-      window.open(
-        'https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97',
-        '_self'
-      );
+  @service router;
+  beforeModel(transition) {
+    if (transition?.to?.queryParams?.dev === 'true') {
+      this.router.transitionTo('new-signup');
+      this.analytics.trackEvent(SIGNUP.PAGE_LOADED);
```

Comment:
I think the analytics call should be before router.transitionTo

Reply:
okay doing it